tempodb/encoding/backend_block.go

```diff
@@ -125,3 +126,50 @@ func (b *BackendBlock) NewIndexReader() (common.IndexReader, error) {
 
 	return reader, nil
 }
+
+func (b *BackendBlock) BlockMeta() *backend.BlockMeta {
+	return b.meta
+}
+
+func (b *BackendBlock) Reader() backend.Reader {
+	return b.reader
+}
+
+func (b *BackendBlock) Write(ctx context.Context, r backend.ReaderReader, w backend.Writer) error {
+	blockID := b.meta.BlockID
+	tenantID := b.meta.TenantID
+
+	copy := func(name string) error {
+		reader, size, err := r.ReadReader(ctx, name, blockID, tenantID)
+		if err != nil {
+			return errors.Wrapf(err, "error reading %s", name)
+		}
+		defer reader.Close()
+
+		return w.WriteReader(ctx, name, blockID, tenantID, reader, size)
+	}
+
+	// Data
+	err := copy(nameObjects)
+	if err != nil {
+		return err
+	}
+
+	// Bloom
+	for i := 0; i < common.GetShardNum(); i++ {
+		err = copy(bloomName(i))
+		if err != nil {
+			return err
+		}
+	}
+
+	// Index
+	err = copy(nameIndex)
+	if err != nil {
+		return err
+	}
+
+	// Meta
+	err = w.WriteBlockMeta(ctx, b.meta)
```

Comment:
why use this backend block's meta and not copy from the reader?

Reply:
The metadata file name and location is actually internal to each backend and handled with a separate `WriteBlockMeta` method.  There is not a wider const for it.  This was discussed during previous refactorings of the backend, and the rationale at the time was that the block metadata is outside any encoding version, so it required a dedicated method.  I think we could create a const value for it and standardize the backends, but it might not be worth adding to this PR.